warehouse/manage/views.py

```diff
@@ -4986,13 +4983,23 @@ def manage_project_history(project, request):
     except ValueError:
         raise HTTPBadRequest("'page' must be an integer.")
 
-    events_query = (
+    project_events_query = (
         request.db.query(Project.Event)
         .join(Project.Event.source)
         .filter(Project.Event.source_id == project.id)
         .order_by(Project.Event.time.desc())
     )
 
+    file_ids = [file.id for release in project.releases for file in release.files]
```

Comment:
Flagging for review: this is a pretty gnarly way to get all associated file events, and there's probably a better way 🙂

Reply:
If we put a project ID column on the event as well, we wouldn't need to do this.